src/org/opensolaris/opengrok/index/Indexer.java

```diff
@@ -101,542 +109,658 @@ private static void A_usage() {
+                    "Files with the named prefix/extension should be analyzed",
+                    "with the given analyzer, where 'analyzer' may be specified",
+                    "using a simple class name (CAnalyzer) or language name (C)",
+                    "(Note, analyzer specification is case sensitive)",
+                    "  Ex: -A .foo:CAnalyzer",
+                    "      will use the C analyzer for all files ending with .foo",
+                    "  Ex: -A bar.:C",
+                    "      will use the C analyzer for all files starting with bar.",
+                    "  Ex: -A .c:-",
+                    "      will disable the c-analyzer for for all files ending with .c").
+                Do( analyzerSpec -> {
+                    String[] arg = ((String)analyzerSpec).split(":");
+                    String fileSpec = arg[0];
+                    String analyzer = arg[1];
+                    configureFileAnalyzer(fileSpec, analyzer);
+                }
+            );
+
+            parser.on("-a", "--leadingWildCards", "=on/off", onOff, Boolean.class, 
+                "Allow or disallow leading wildcards in a search.").Do( v -> {
+                cfg.setAllowLeadingWildcard((Boolean)v);
+            });
+
+            parser.on("-B", "--infoProvider", "=URL",
+                "Base URL of the user Information provider.",
+                "Example: \"http://www.myserver.org/viewProfile.jspa?username=\".",
+                "Use \"none\" to disable link.").Do( v -> {
+                cfg.setUserPage((String)v);
+            });
+
+            parser.on("-C", "--progress",
+                "Print per project percentage progress information",
+                "(I/O extensive, since one read through directory structure is",
+                "made before indexing, needs -v, otherwise it just goes to the log)").
+                Do( v -> {
+                    cfg.setPrintProgress(true);
+                }
+            );
 
-                /*
-                 * Add paths to directories under source root. If projects
-                 * are enabled the path should correspond to a project because
-                 * project path is necessary to correctly set index directory
-                 * (otherwise the index files will end up in index data root
-                 * directory and not per project data root directory).
-                 * For the check we need to have 'env' already set.
-                 */
-                for (String path : subFilesList) {
-                    String srcPath = env.getSourceRootPath();
-                    if (srcPath == null) {
-                        System.err.println("Error getting source root from environment. Exiting.");
-                        System.exit(1);
+            parser.on("-c", "--ctags","=/path/to/ctags", 
+                "Path to Exuberant Ctags from http://ctags.sf.net",
```

Comment:
- say Exuberant or Universal ctags
- remove the http link

Reply:
done